results-processor/processor.py

```diff
@@ -187,13 +188,14 @@ def _download_archive(self, archive_url):
             return
         with zipfile.ZipFile(artifact, mode='r') as z:
             for f in z.infolist():
-                # ZipInfo.is_dir isn't available in Python 3.5.
-                if f.filename.endswith('/'):
+                if f.is_dir():
                     continue
-                path = z.extract(f, path=self._temp_dir)
-                if re.match(r'^.*/wpt_report.*\.json$', f.filename):
+                basename = posixpath.basename(f.filename)
+                if fnmatch.fnmatchcase(basename, 'wpt_report*.json'):
+                    path = z.extract(f, path=self._temp_dir)
                     self.results.append(path)
-                if re.match(r'^.*/wpt_screenshot.*\.txt$', f.filename):
+                elif fnmatch.fnmatchcase(basename, 'wpt_screenshot*.txt'):
+                    path = z.extract(f, path=self._temp_dir)
```

Comment:
Why did you move (and duplicate) the path assignment into the conditional? Do the if/elif conditions have side effects?

Reply:
Mostly just avoiding decompressing things unnecessarily, try to mitigate some of the risk of any sort of decompression bomb by only decompressing the files we actually care about.